jvm/src/test/scala/scala/xml/XMLTest.scala

```diff
@@ -667,13 +667,14 @@ class XMLTestJVM {
   @UnitTest
   def checkThatErrorHandlerIsNotOverwritten(): Unit = {
     var gotAnError: Boolean = false
-    XML.reader.setErrorHandler(new org.xml.sax.ErrorHandler {
+    val reader = XML.reader
+    reader.setErrorHandler(new org.xml.sax.ErrorHandler {
       override def warning(e: SAXParseException): Unit = gotAnError = true
       override def error(e: SAXParseException): Unit = gotAnError = true
       override def fatalError(e: SAXParseException): Unit = gotAnError = true
     })
     try {
-      XML.loadString("<a>")
+      XML.adapter.loadDocument(Source.fromString("<a>"), reader)
```

Comment:
I'd love to make `XMLLoader.adapter`, `NoBindingFactoryAdapter` and `FactoryAdapter` private at some point (and remove some of them altogether), and add `XMLLoader.loadDocument(inputSource: InputSource, reader: XMLReader)`...